typegen/strats/inline.py

```diff
@@ -101,67 +103,104 @@ def visit_FunctionDef(self, fdef: ast.FunctionDef):
+                    relevant_trace_data_class_member[TraceData.VARNAME] == target_name]
+            elif isinstance(target_node, ast.Name):
+                # Finds the local variable type hint.
+                target_trace_data = relevant_trace_data_local_variables[
+                    relevant_trace_data_local_variables[TraceData.VARNAME] == target_name]
+            if target_trace_data is None or len(target_trace_data) == 0:
+                logger.debug(
+                    f"No hint found for assign for '{target_name}'"
+                )
+                continue
+
             logger.debug(
-                f"Applying type hints for simple assignment '{node.targets[0].id}'"
-            )
-            return ast.AnnAssign(
-                target=node.targets[0],
-                value=node.value,
-                annotation=ast.Name(node_vars[TraceData.VARTYPE].values[0]),
-                simple=True,
+                f"Applying type hints for simple assignment '{target_name}'"
             )
-
-        else:
-            prehints = list(
-                node_vars[[TraceData.VARNAME, TraceData.VARTYPE]].itertuples(
-                    index=False, name=None
+            if contains_one_target:
+                new_node = ast.AnnAssign(
+                    target_node,
+                    value=node.value,
+                    annotation=ast.Name(target_trace_data[TraceData.VARTYPE].values[0]),
+                    simple=True,
                 )
-            )
-            logger.debug(f"Applying type hints for multi-assignments '{prehints}'")
-
-            assigns = [
-                ast.AnnAssign(
-                    target=ast.Name(name), annotation=ast.Name(hint), simple=True
+                return new_node
+            else:
+                new_node = ast.AnnAssign(
+                    target_node,
+                    annotation=ast.Name(target_trace_data[TraceData.VARTYPE].values[0]),
+                    simple=True
                 )
-                for name, hint in prehints
-            ]
 
-            newhints: list[ast.AST] = list()
-            newhints.extend(assigns)
-            newhints.append(node)
+            new_nodes.append(new_node)
 
-            return newhints
+        new_nodes.append(node)
+        return new_nodes
 
-    def _extract_assign_ids(self, node: ast.AST) -> list[str]:
-        # https://stackoverflow.com/a/72231602
-        return [i.id for i in ast.walk(node) if isinstance(i, ast.Name)]
+    def _extract_target_names_with_nodes(self, node: ast.AST) -> list[typing.Tuple[str, ast.Name | ast.Attribute]]:
+        target_names_with_nodes: list[typing.Tuple[str, ast.Name | ast.Attribute]] = []
+        if isinstance(node, ast.Attribute):
+            target_names_with_nodes.append((node.attr, node))
+        elif isinstance(node, ast.Name):
+            target_names_with_nodes.append((node.id, node))
+        else:
+            for child_node in ast.iter_child_nodes(node):
```

Comment:
consider using `ast.walk` instead to do the recursion for us

Reply:
The child nodes of a node should not be checked if the node itself is added to the list. Otherwise, check for each child of the node whether the same applies to this child. The reason for doing this is that no nodes containing the self value are selected as targets, which are the child nodes of class members nodes. I can not think of any implementation which uses ast.walk and achieve the same functionality.

Reply:
Ok, then just document this with a comment.

Reply:
Done

Reply:
Done